### PR TITLE
added conditional SystemD support for notify

### PR DESCRIPTION
**Pull request by expeditioneer**
This adds SystemD notify support.

Based on patch of Jakub Jelen <jjelen@redhat.com> from RedHat / CentOS packaging.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -4693,6 +4693,29 @@ AC_ARG_WITH([kerberos5],
...
+	[  --with-systemd          Enable systemd support],
+	[ if test "x$withval" != "xno" ; then
+		AC_PATH_TOOL([PKGCONFIG], [pkg-config], [no])
+		if test "$PKGCONFIG" != "no"; then
+			AC_MSG_CHECKING([for libsystemd])
+			if $PKGCONFIG --exists libsystemd; then
+				SYSTEMD_CFLAGS=`$PKGCONFIG --cflags libsystemd`
+				SYSTEMD_LIBS=`$PKGCONFIG --libs libsystemd`
+				CPPFLAGS="$CPPFLAGS $SYSTEMD_CFLAGS"
+                              +				SSHDLIBS="$SSHDLIBS $SYSTEMD_LIBS"
```
Comment (elarsen-meazure) [inline configure.ac:4708]:
```suggestion
				SSHDLIBS="$SSHDLIBS $SYSTEMD_LIBS"
```
I had to remove this superfluous `+` character to get the SSHDLIBS popluted correctly.

**Review by elarsen-meazure — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 1 above)
Comment (expeditioneer) [reply configure.ac]: I've fixed this.

**Comment by spacexnasa** [comment]
hello,@expeditioneer, i tested openssh-9.4p1 centos 7,cann't work,if you hava time,can make a test ?

**Comment by djmdjm** [comment]
We're not interested in taking a dependency on libsystemd, see https://bugzilla.mindrot.org/show_bug.cgi?id=2641 for the approach we are likely to take.

**Pull request closed without merging**